src/overlays/actors/ovl_En_Snowman/z_en_snowman.h

```diff
@@ -2,16 +2,56 @@
 #define Z_EN_SNOWMAN_H
 
 #include "global.h"
+#include "objects/object_snowman/object_snowman.h"
+
+#define EN_SNOWMAN_GET_TYPE(thisx) ((thisx)->params)
+#define EN_SNOWMAN_GET_ATTACK_RANGE(thisx) (((thisx)->params >> 8) & 0xFF)
+
+typedef enum {
+    /* 0 */ EN_SNOWMAN_TYPE_SMALL,
+    /* 1 */ EN_SNOWMAN_TYPE_LARGE,
+    /* 2 */ EN_SNOWMAN_TYPE_SPLIT,
+    /* 3 */ EN_SNOWMAN_TYPE_SMALL_SNOWBALL,
+    /* 4 */ EN_SNOWMAN_TYPE_LARGE_SNOWBALL,
+} EnSnowmanType;
 
 struct EnSnowman;
 
 typedef void (*EnSnowmanActionFunc)(struct EnSnowman*, GlobalContext*);
 
 typedef struct EnSnowman {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x13C];
-    /* 0x0280 */ EnSnowmanActionFunc actionFunc;
-    /* 0x0284 */ char unk_284[0xF4];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ SkelAnime skelAnime;
+    /* 0x188 */ SkelAnime snowPileSkelAnime;
+    /* 0x1CC */ Vec3s jointTable[EENO_LIMB_MAX];
+    /* 0x214 */ Vec3s morphTable[EENO_LIMB_MAX];
+    /* 0x25C */ Vec3s snowPileJointTable[EENO_SNOW_PILE_LIMB_MAX];
+    /* 0x26E */ Vec3s snowPileMorphTable[EENO_SNOW_PILE_LIMB_MAX];
+    /* 0x280 */ EnSnowmanActionFunc actionFunc;
+    /* 0x284 */ EnSnowmanActionFunc oldActionFunc;
+    /* 0x288 */ u8 isHoldingSnowball;
+    /* 0x289 */ u8 combineState;
+    /* 0x28A */ u8 turningOnSteepSlope;
+    /* 0x28B */ u8 drawDmgEffType;
+    /* 0x28C */ union {
+                    s16 timer;
+                    s16 snowballsToThrowBeforeIdling;
+                };
+    /* 0x28E */ s16 snowPileTargetRotY;
+    /* 0x290 */ s16 combineTimer;
+    /* 0x294 */ f32 eenoScale;
+    /* 0x298 */ union {
+                    f32 frameToStartHoldingSnowball;
+                    f32 frameToThrowSnowball;
+                    f32 targetScaleDuringCombine;
+                } work;
```

Comment:
Usually fwork for floats... indeed, I'm not sure we usually use work for unions. Also, I think this needs to telegraph better that it holds 3 distinct things. Yes it's the source of a glitch, but we need to signpost it so that people don't make the same mistake if modding!

Reply:
When you say "telegraph better", do you mean changing the name of `fwork` to something else? Or do you mean some sort of comment? I'm not sure what I would name this that would better signal it holds distinct things.

Reply:
I think we'll have to live with a named union for now. As I said below, defines are an option as well, but best to make a project-wide decision on that.

Reply:
Yeah, it's unfortunate that OoT hasn't decided on a solution for this yet, but whatever it is, we should be consistent. I'll leave this (and the other one) as-is for now.